### PR TITLE
[sinon-chrome] Support plugins

### PR DESCRIPTION
**Pull request by kobanyan**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/acvetkov/sinon-chrome#plugins
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/sinon-chrome/index.d.ts [hunk 1]
```diff
@@ -353,6 +356,23 @@ declare namespace SinonChrome.permissions {
...
+    export interface Plugin {}
+    export interface Translation {
+        message: string;
+        placeholders?: object;
+    }
+    export interface Translations {
+        [key: string]: Translation;
+    }
+    export class I18nPlugin implements Plugin {
+        constructor(translations: Translations);
```
Comment (pimterry) [inline types/sinon-chrome/index.d.ts]: Looks like this parameter is actually optional too

File: types/sinon-chrome/index.d.ts [hunk 2]
```diff
@@ -353,6 +356,23 @@ declare namespace SinonChrome.permissions {
     export var request: SinonChromeStub;
 }
 
+declare namespace SinonChrome.plugins {
+    export interface Plugin {}
```
Comment (pimterry) [inline types/sinon-chrome/index.d.ts]: Implementing this (in the plugins below) doesn't do anything at all, and requiring for it as the type for the `registerPlugin` param doesn't do anything beyond `plugin: {}`, which doesn't seem useful.

We could drop this, and just use `plugin: {}` above, or change this to be a type enumeration limited to these existing plugins (`export type Plugin = I18nPlugin | CookiePlugin`) or change the types here to properly represent what's required, which I think is something like `plugin: { install: (chrome: SinonChrome) => void }` (based on https://github.com/acvetkov/sinon-chrome/blob/master/src/api/index.js#L47-L57). What do you think?

**Review by pimterry — CHANGES_REQUESTED** [review]

File: types/sinon-chrome/index.d.ts [hunk 3]
```diff
@@ -353,6 +356,23 @@ declare namespace SinonChrome.permissions {
...
+    export interface Translation {
```
Comment (crimx) [inline types/sinon-chrome/index.d.ts]: Maybe follow the browser spec:

```javascript
export interface Translation {
  message: string;
  description?: string;
  placeholders?: {
    [key: string]: {
      content: string,
      example?: string,
    }
  };
}
```

**Review by crimx — CHANGES_REQUESTED** [review]

File: types/sinon-chrome/index.d.ts (hunk 2 above)
Comment (kobanyan) [reply types/sinon-chrome/index.d.ts]: Thanks for the feedback.
I make it optional.

File: types/sinon-chrome/index.d.ts (hunk 3 above)
Comment (kobanyan) [reply types/sinon-chrome/index.d.ts]: Thanks for the feedback.
I follow the browser spec.

File: types/sinon-chrome/index.d.ts (hunk 1 above)
Comment (kobanyan) [reply types/sinon-chrome/index.d.ts]: Thanks for the feedback.
I make it optional.

**Review by crimx — APPROVED** [review]

**Review by pimterry — APPROVED** [review]

**Pull request merged**